### PR TITLE
Only sync GSD files when writing out properties rather than per property

### PR DESCRIPTION
**Pull request by vyasr**
## Description

See glotzerlab/hoomd-blue#499 for more information.

## Change log

```
Update fsync calls for namelist to only call after all names are added to the list.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/gsd/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**GSD Contributor Agreement**](https://github.com/glotzerlab/gsd/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/gsd/blob/master/doc/credits.rst).

### CODE REVIEW

**Comment by vyasr** [comment]
Windows doesn't seem to like my changes, but I'm not sure why since the fsync calls already existed and I'm just changing when they are called.

File: gsd/gsd.c [hunk 1]
```diff
@@ -630,7 +628,7 @@ uint32_t gsd_make_version(unsigned int major, unsigned int minor)
 int gsd_create(const char *fname, const char *application, const char *schema, uint32_t schema_version)
     {
     int extra_flags = 0;
-    #ifdef _WIN32
```
Comment (joaander) [inline gsd/gsd.c]: Revert this change and the similar ones below to fix the failing test. This was fixed in #51.

**Review by joaander — CHANGES_REQUESTED** [review]

**Review by joaander — APPROVED** [review]
Thanks!

**Pull request merged**